src/main/scala/de/mukis/sbt/slick/Keys.scala

```diff
@@ -19,12 +20,13 @@ trait Keys {
 
   val slickDriver = SettingKey[String]("slick-driver", "The slick driver, e.g. 'scala.slick.driver.MySQLDriver'")
   val slickJDBCDriver = SettingKey[String]("slick-jdbc-driver", "The jdbc driver, e.g. 'com.mysql.jdbc.Driver'")
-  val slickUrl = SettingKey[MakeUrl]("slick-url", "URL to database, e.g. 'jdbc:mysql://localhost:3306/your-db'")
+  val slickUrl = TaskKey[MakeUrl]("slick-url", "URL to database, e.g. 'jdbc:mysql://localhost:3306/your-db'")
+  val slickArguments = TaskKey[MakeArguments]("slick-arguments", "Arguments array for database")
```

Comment:
By introducing the `slickArguments` every user can break the call order of the code-gen main.

I would recommend to use the `slickUser` and `slickPassword` and append in in the `SlickCodeGenPlugin` to the arguments.